bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java

```diff
@@ -200,7 +200,12 @@ public void run() {
                     socket.setReuseAddress(true);
                     while (!stopped) {
                         DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
-                        socket.receive(packet);
+                        if (!socket.isClosed()) {
+                            socket.receive(packet);
+                            logger.warn("Socket was unexpectedly closed");
```

Comment:
Not sure if i understand this as 2 lines before: the is a check if socket is not closed.  
I would rather expecct somethingt like this:
```
                        if (!socket.isClosed()) {
                            socket.receive(packet);
                        } else {
                            logger.warn("Socket was unexpectedly closed");
                        }

```

Reply:
Oh yeah, you are right — I should probably to to bed and not review any more code 😉

Reply:
if-statements are really difficult constructs ;-) and you should do this while being on a OH booth :D Thanks, @lsiepel